src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala

```diff
@@ -81,19 +82,26 @@ object EasyArchiveBag extends Bagit5FacadeComponent with DebugEnhancedLogging {
   }
 
 
-  private def getBagSequence(bagId: BagId)(implicit ps: Parameters): Try[Option[String]] = {
-    Try {
-      val http = createHttpClient(ps.bagIndexService.getHost, ps.bagIndexService.getPort, "", "")
-      val get = new HttpGet(ps.bagIndexService.resolve(s"bag-sequence?contains=$bagId"))
-      get.addHeader("Accept", "text/plain;charset=utf-8")
-      val sw = new StringWriter()
-      val response = http.execute(get)
-      if (response.getStatusLine.getStatusCode != HttpStatus.SC_OK) throw new IllegalStateException(s"Error retrieving bag-sequence for bag: $bagId")
-      IOUtils.copy(response.getEntity.getContent, sw, "UTF-8")
-      Some(sw.toString)
+  private def getBagSequence(bagId: BagId)(implicit ps: Parameters): Try[Option[String]] = Try {
+    val http = createHttpClient(ps.bagIndexService.getHost, ps.bagIndexService.getPort, "", "")
+    val get = new HttpGet(ps.bagIndexService.resolve(s"bag-sequence?contains=$bagId"))
+    get.addHeader("Accept", "text/plain;charset=utf-8")
+    val sw = new StringWriter()
+    val response = http.execute(get)
+    val statusLine = response.getStatusLine
+    if (statusLine.getStatusCode != HttpStatus.SC_OK)
+      throwBagNotFoud(bagId, s"bag index [${get.getURI}] returned ${statusLine.getStatusCode} ${statusLine.getReasonPhrase}")
+    IOUtils.copy(response.getEntity.getContent, sw, "UTF-8")
+    sw.toString match {
+      case s if s.isBlank => throwBagNotFoud(bagId, s"is not found in bag index [${get.getURI}]")
```

Comment:
Note that the URI duplicates the bag-id of line 102, example of a message:

    Exception in thread "main" java.lang.IllegalStateException: Bag with bag-id 9a840d17-103a-46e3-99a6-abf7ea47642c, pointed to by Is-Version-Of field in bag-info.txt, is not found in bag index [http://localhost:20120/bag-sequence?contains=9a840d17-103a-46e3-99a6-abf7ea47642c]

Reply:
throwBagNotFoud --> throwBagNotFound